File: api/collegeportal/src/main/java/com/collegeportal/modules/admin/service/AdminService.java
package com.collegeportal.modules.admin.service;

import com.collegeportal.modules.admin.dto.response.AdminResponseDTO;
import com.collegeportal.modules.admin.dto.response.AdminStatsDTO;
import com.collegeportal.shared.enums.RoleType;

import java.util.List;

public interface AdminService {
    List<AdminResponseDTO> getPendingUsers(RoleType role);
    List<AdminResponseDTO> getApprovedUsers(RoleType role);
    List<AdminResponseDTO> getRejectedUsers();
    AdminResponseDTO approveUser(Long userId);
    void rejectUser(Long userId, String reason);
    void revokeUser(Long userId);
    void deleteUser(Long userId);
    void bulkApprove(List<Long> userIds);
    void bulkReject(List<Long> userIds, String reason);
    AdminStatsDTO getStats();
}

File: web/src/services/adminService.js
import axiosInstance from '../api/axiosInstance';

const BASE = '/api/v1/admin';

export const getStats = () => axiosInstance.get(`${BASE}/stats`).then((r) => r.data);
export const getPendingUsers = () => axiosInstance.get(`${BASE}/pending-users`).then((r) => r.data);
export const approveUser = (userId) => axiosInstance.put(`${BASE}/approve/${userId}`).then((r) => r.data);
export const rejectUser = (userId) => axiosInstance.put(`${BASE}/reject/${userId}`).then((r) => r.data);
